Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1603,6 +1634,12 @@ void GridTrackSizingAlgorithm::run()
     ASSERT(wasSetup());
     StateMachine stateMachine(*this);
 
+    if (m_renderGrid->areMasonryRows() && m_direction == ForRows)
```

Comment:
Adding a helper for `RenderGrid::isMasonry(GridTrackSizingDirection)` would be nice, so that these two checks could just be `if (m_renderGrid->isMasonry(m_direction)) return;`